ci-operator/step-registry/openshift/microshift/e2e/metal-tests/openshift-microshift-e2e-metal-tests-commands.sh

```diff
@@ -20,10 +20,12 @@ EOF
 chmod 0600 "${HOME}/.ssh/config"
 
 trap 'scp -r ${INSTANCE_PREFIX}:/home/${HOST_USER}/microshift/_output/test-images/scenario-info ${ARTIFACT_DIR}' EXIT
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
```

Comment:
I think there may be a race here. Between the time when `jobs -p` is run and until we execute the `kill` command, some of the processes may terminate. This would cause `kill` to return `false` and `wait` would not be called.

Reply:
Should we call wait unconditionally?